### PR TITLE
Make the relation bobble follow the mouse on drag

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5584/
Test FAILed.

**Comment by anthonydillon** [comment]
## QA
- Add two relatable services onto the canvas
- Hover over one and drag the relation icon over to the other
- The relation icon should follow the mouse with the pending relation
- Hover over the other service and see that it snaps and the relation icon hides
- Create the relation
- Try create another relation and the relation line should hide and the other one fade out as it has a relation already
- Try click relate, make sure that still works

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5585/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5586/
Test PASSed.

**Comment by hatched** [comment]
QA Notes:

This is such a cool UI feature :) Thanks for implementing it

Issue I had was that if I 'clicked' on the icon like was required previously the icon disappears for a while then appears.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5601/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5603/
Test PASSed.

File: jujugui/static/gui/src/app/views/topology/relation.js [hunk 1]
```diff
@@ -114,7 +114,7 @@ YUI.add('juju-topology-relation', function(Y) {
           @event fade
           @param {Object} An object with a service name to hide
         */
-        fade: 'fade',
+        fade: 'hide',
```
Comment (hatched) [inline jujugui/static/gui/src/app/views/topology/relation.js]: what's going on here?

File: jujugui/static/gui/src/test/test_environment_view.js [hunk 2]
```diff
@@ -1003,15 +1003,6 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
                    'service does not have the ' + cssClass + ' class');
     }
 
-    it('should fade services and relations', function(done) {
```
Comment (hatched) [inline jujugui/static/gui/src/test/test_environment_view.js]: Do we not fade anymore?

**Comment by hatched** [comment]
QA Issue:

If I click the build relation icon, move the mouse to the available applications and then click the application, it doesn't create the relation. The only way I can create the relation is by click&drag to the other service.

File: jujugui/static/gui/src/app/views/topology/relation.js (hunk 1 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/views/topology/relation.js]: We are now fading a unrelatable service but we want the relation line to hide.

File: jujugui/static/gui/src/test/test_environment_view.js (hunk 2 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/test/test_environment_view.js]: We do fade but this check is to test the relation line and the service fades in a linked way but that is no longer that case.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5640/
Test PASSed.

**Comment by anthonydillon** [comment]
@hatched thanks, fixed up the click relation and QA'ed on GI Joe.

**Comment by hatched** [comment]
Thanks for keeping at this @anthonydillon it's an awesome UX

:+1: :ok:

**Comment by makyo** [comment]
👍

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**